### PR TITLE
clarify instructions to use array for players and add tests

### PR DESCRIPTION
**Pull request by realAndrewCohn**
the readme asks students to create a hash of players with player names as keys. An array of all characters would be more intuitive and the solution branch uses an array of players. I've updated the readme and added tests. This issue is address in #30

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +121,8 @@ end
 
 Open up the `hashketball.rb` file and add the line *require "pry"* at the top and copy and paste the above method. Then, beneath the `end` that closes the method definition, call the method (`good_practices`) and, in your terminal, run the file with `ruby hashketball.rb`. Play around with the methods in each binding until you get comfortable with the iteration. This should give you a stronger sense of how we iterate through so many levels of a nested hash and what happens on each level. **Use this method of placing LOTS of bindings when you iterate in order to solve this lab.**
 
+*If you are getting errors when attempting to use pry, make sure you call* `gem install pry` in your terminal.
+
 Okay, *now* we're ready to build out methods:
 
 ### Method Building
```
Comment (nikymorg) [inline README.md]: Should we add name into the expected response for player_stats on 144? This would need to be updated in the tests as well.

Comment (realAndrewCohn) [reply README.md]: yes that sounds good––i'll update

Comment (realAndrewCohn) [reply README.md]: this particular method should take in a player name and return all of their data _except_ for their name

Comment (realAndrewCohn) [reply README.md]: we could change it as i'm not sure asking students to remove a key/value pair from a hash is particularly useful

Comment (DanielSeehausen) [reply README.md]: not sure we are on the right track here -- see comment on pull request. separately:

a method called `player_stats` that expects a name and returns stats makes sense from my perspective. the name is already known (its whats passed), its the stats we want. This change is in response to the hash --> array change, which is addressed in comment on pr.

**Comment by DanielSeehausen** [comment]
going to request amending this for the following reasons:
- lab is to teach hashes, and we want them working with hash access/mindset. Swapping a hash to an array prevents this
- unless i'm misunderstanding, using a hash instead of an array makes sense in terms of storing this information. we want instant lookup, not traversal, for a specific player's information
- while name is not good value to be hashing as a key (because of potential for collisions), students aren't focusing on details like that at this point

To offer a potential solution, and in building off of the issue raised by alex and this pr, I would propose the following:
- the solution is made to match the readme ( players => {}, not players => [] )
- either:
   - the specs expect name to be including in control responses, and the solution accurately adds k/v name => name into the hash
   - the specs don't expect name to be included in control responses, and the solution can simply return the value from players["name"]

**Comment by realAndrewCohn** [comment]
Copying the conversation from Slack so we have it here for reference: changing it to `players => {}` makes more sense. The purpose of the lab is hash traversal but does that also mean we want to keep arrays out of this as much as possible? I’d argue that dealing with a hash containing arrays is good practice for them. On the other hand, `#player_stats` should just return an individual player’s stats and it’s awkward to have to delete a k/v pair from a hash, which would be the case with an array of player hashes. I’ll update the code in the solution branch and submit a pr

**Pull request closed without merging**